### ISSUE TITLE
'workspace' section overrides base folder entirely

### ISSUE THREAD

**Issue by asaf** [drone/drone#2202]
Hey,

I have a docker image with some folders under `/go`

```bash
ls /go
bin
pkg
src
```

If in my drone manifest I put the `workspace` section such:

```yml
workspace:
  base: /go
  path: src/bitbucket.org/my/project
```

Then the `/go` folder gets overridden entirely and contains only the `/go/src` folder.
```bash
ls /go
src
```

Drone should not remove the folder rather put the workspace's path under the base folder.

(If I remove the 'workspace' section everything work as expected)

Thanks.

**Comment by asaf** [comment]
Well, I guess this is due to the 'volume sharing' behavior,

I workaround this by changing the base to `/go/src` such

```
workspace:
  base: /go/src
  path: bitbucket.org/indexia/crossid
```

I'm keeping this issue open as maybe it worth mentioning this behavior in the docs.

thanks.

**Comment by enjoylife** [comment]
@bradrydzewski,  I think this issue is still relevant. 
My reasoning is thus: currently the [docs](http://readme.drone.io/0.5/usage/customize-workspace/) on workspaces have this helpful "hello world for go" pipeline:

```
# drone.yaml
workspace:
   base: /go
   path: src/github.com/octocat/hello-world
pipeline:
  build:
    image: golang:latest
    commands:
      - go get
      - go test
```

Which works, and is simple. But if someone was to swap out the image with a custom docker file and add a few more go related commands:
```
# Dockerfile golang-custom
RUN go get -u  github.com/alecthomas/gometalinter; gometalinter --install   
```

```
# drone.yaml
commands:
      - go get
      - gometalinter ...
```
We arrive at the unexpected behavior of the gometalinter binary and tools not being found. Which makes sense retrospectively as @asaf  and myself discovered since the basepath and go bin is overridden.

But if we put ourselves into the mindset of a newcomer to the drone ecosystem, its not immediately apparent that the workspace is causing the issue. When making changes to the dockerfile and adding commands, a newcomer isn't exactly thinking about workspaces. The workspace isn't thought about much since the workspaces option is more of a set it and forget it kind of configuration.

Instead newcomers such as myself might go debugging down dead-ends thinking its an issue in their build command or their dockerfile, etc. 

In short, a helpful change to the docs might be:

```
workspace:
    + base: /godev
```
Or at the least, it might be relevant to document this overriding behavior on the [docs](http://readme.drone.io/0.5/usage/customize-workspace/). This potentially will help users make better initial decisions when configuring their drone workspace.
